src/adf_dir.c

```diff
@@ -225,7 +225,8 @@ RETCODE adfRemoveEntry(struct Volume *vol, SECTNUM pSect, char *name)
 
     if (entry.secType==ST_FILE) {
         adfFreeFileBlocks(vol, (struct bFileHeaderBlock*)&entry);
-        if (adfEnv.useNotify)
+    	adfSetBlockFree(vol, nSect); //marks the FileHeaderBlock as free in BitmapBlock
```

Comment:
you should avoid gratuitous whitespace changes such as this one (here, indentation with spaces was changed to a mix of tab and spaces)